pkg/lib/release/release.go

```diff
@@ -0,0 +1,62 @@
+package release
+
+import (
+	"encoding/json"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+)
+
+// +k8s:openapi-gen=true
+// OperatorRelease is a wrapper around a slice of semver.PRVersion which supports correct
+// marshaling to YAML and JSON.
+// +kubebuilder:validation:Type=string
```

Comment:
We should add more validation here:
- CEL expression that does a regex match to make sure it is a valid prerelease string
- Max length (keeping in mind that we expect the release to be included in the CSV metadata.name, which has its own length restrictions.

Was there any API review on this front? They'd be better at pointing out all the things.

Reply:
Added some new CEL validations, including a max length of 20 chars.  This was a nice, round number which was big enough for all known uses of analogous features (Freshmaker) as well as a minimal-precision date/time-stamp, and should leave ~40 chars to remain valid when represented as a string suffix joined with the CSV.name and CSV.version as the metadata.name of the bundle with a release version (plus connecting characters). 

Also added to the bundle validators to enforce the case where the author provides a noncompliant name for the release version case.

Reply:
Any chance you looked for bundles that have the `olm.substitutesFor` annotation and checked to see if the overall `<package-name>-v<version>-<release-version>` format that we would require if they released again using a release field that followed the format of their version build metadata would overflow the `metadata.name` length requirement of a CSV?